clients/da-vinci-client/src/main/java/com/linkedin/davinci/store/view/MaterializedViewWriter.java

```diff
@@ -41,13 +39,12 @@ public MaterializedViewWriter(
       Schema keySchema,
       Map<String, String> extraViewParameters) {
     super(props, version, keySchema, extraViewParameters);
-    pubSubProducerAdapterFactory = props.getVeniceServerConfig().getPubSubClientsFactory().getProducerAdapterFactory();
     internalView =
         new MaterializedView(props.getCombinedProperties().toProperties(), version.getStoreName(), extraViewParameters);
     materializedViewTopicName =
         internalView.getTopicNamesAndConfigsForVersion(version.getNumber()).keySet().stream().findAny().get();
     this.veniceWriter = Lazy.of(
-        () -> new VeniceWriterFactory(props.getCombinedProperties().toProperties(), pubSubProducerAdapterFactory, null)
+        () -> new VeniceWriterFactory(props.getCombinedProperties().toProperties(), null)
```

Comment:
same: Let’s avoid creating factories at individual call sites. Instead, we should create the factory
once per server component and reuse it.